### PR TITLE
IOSim versions of Server2 tests

### PR DESCRIPTION
**Pull request by UlfNorell**

### CODE REVIEW

**Comment by coot** [comment]
@UlfNorell if you rebase the CI should become green.

**Comment by coot** [comment]
@UlfNorell sorry you'll need rebase again, we pushed today a `p2p-master` branches based on `1.27.0` release of `cardano-node`.

File: ouroboros-network-framework/test/Test/Ouroboros/Network/IOSim.hs [hunk 1]
```diff
@@ -0,0 +1,654 @@
...
+
+prop_shrinker_BearerInfoScript :: Fixed BearerInfoScript -> Bool
+prop_shrinker_BearerInfoScript (Fixed bis) =
+    all (\bis'@(BearerInfoScript (Script s)) ->
+                  bis /= bis'
+               && not (canFail (NonEmpty.last s))
+        )
+        (shrink bis)
+
+newtype NonFailingBearerInfoScript = NonFailingBearerInfoScript (Script AbsBearerInfo)
```
Comment (coot) [inline ouroboros-network-framework/test/Test/Ouroboros/Network/IOSim.hs:598]:
```suggestion
newtype NonFailingBearerInfoScript = 
    NonFailingBearerInfoScript (Script AbsBearerInfo)
```

File: ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs [hunk 2]
```diff
@@ -926,17 +926,17 @@ bidirectionalExperiment
 
               pure $
                 foldr
-                  (\(r, expected) acc ->
+                  (\ (r, expected) acc ->
                     case r of
-                      Left _ -> acc
+                      Left err -> counterexample (show err) False
```
Comment (coot) [inline ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs:956]: :+1:

File: ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs [hunk 3]
```diff
@@ -144,6 +144,13 @@ instance ( Arbitrary req
                           <*> arbitraryList
                           <*> arbitraryList
 
+    shrink (ClientAndServerData fun ini hot warm est) = concat
+      [ shrink fun  <&> \ fun'  -> ClientAndServerData fun' ini  hot  warm  est
+      , shrink ini  <&> \ ini'  -> ClientAndServerData fun  ini' hot  warm  est
+      , shrink hot  <&> \ hot'  -> ClientAndServerData fun  ini  hot' warm  est
+      , shrink warm <&> \ warm' -> ClientAndServerData fun  ini  hot  warm' est
+      , shrink est  <&> \ est'  -> ClientAndServerData fun  ini  hot  warm  est'
+      ]
```
Comment (coot) [inline ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs:161]: :+1:

**Review by coot — COMMENTED** [review]
LGTM, just minor stylistic changes.

**Comment by UlfNorell** [comment]
It looks like my commits are already on `coot/io-sim-netenv`. Should I rebase and keep the PR open for the cosmetic changes?

**Comment by coot** [comment]
> It looks like my commits are already on coot/io-sim-netenv.
Are you sure? The `coot/io-sim-netenv` branch is at aaed567aaa85695a8d760b5383b798cb7490787f right now.
> Should I rebase and keep the PR open for the cosmetic changes?
Squash the cosmetic changes onto right commits, and I'll merge it to `coot/io-sim-netenv`.

**Comment by UlfNorell** [comment]
Oh no you're right, that was my local copy. My bad.

**Pull request closed without merging**